libraries/AP_Scripting/AP_Scripting.h

```diff
@@ -28,9 +28,10 @@ class AP_Scripting
     AP_Scripting(const AP_Scripting &other) = delete;
     AP_Scripting &operator=(const AP_Scripting&) = delete;
 
-    bool init(void);
+    void init(void);
+    bool init_failed(void) const { return _init_failed; }
 
-    bool enabled(void) const { return _enable != 0; };
+    bool enabled(void) const { return (_enable != 0) && !init_failed(); };
```

Comment:
Do we really want to hook this here? In a lot of ways you want to be able to distinguish between "does the user want this to run right now" and "we wanted to run, but can't". As far as I can tell this isn't actually needed for this PR to do it's desired job.

Reply:
I've reverted this change, thanks.